p2panda-net/src/sync/manager.rs

```diff
@@ -308,18 +308,14 @@ where
 
         if let Some(err) = err.downcast_ref() {
             match err {
+                // If the sync attempt failed because of a connection error we want to retry up to
+                // `max_retry_attempts`. If error occurs after this we simply stop trying without
+                // informing the engine as it never knew the attempts were occurring.
                 SyncAttemptError::Connection => {
                     warn!("sync attempt failed due to connection error");
                     if sync_attempt.attempts <= self.config.max_retry_attempts {
                         self.reschedule_attempt(sync_attempt).await?;
                         return Ok(());
-                    } else {
```

Comment:
Notifying the engine of `SyncFailed` here was causing a panic when the gossip buffer was unlocked as it was never locked in the first place. @mycognosist @adzialocha do you mind confirming that this doesn't have any other repercussions I'm not aware of?

Reply:
Yes! I stumbled upon it as well but wasn't sure.

On a connection fail the session never starts, but a `SyncFailed` is sent anyhow, so indeed it should be removed (or treated differently by the engine).